packages/statemanager/test/stateManager.account.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { Address, KECCAK256_RLP, bytesToHex, equalsBytes, hexStringToBytes } from '@ethereumjs/util'
+import * as tape from 'tape'
+// explicitly import `inherits` to fix karma-typescript issue
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { inherits } from 'util'
+
+import { DefaultStateManager } from '../src'
+
+import { createAccount } from './util'
+
+tape('StateManager -> General/Account', (t) => {
+  for (const accountCacheOpts of [{ deactivate: false }, { deactivate: true }]) {
+    t.test('should instantiate', async (st) => {
+      const stateManager = new DefaultStateManager({ accountCacheOpts })
+
+      st.deepEqual(stateManager._trie.root(), KECCAK256_RLP, 'it has default root')
+      const res = await stateManager.getStateRoot()
+      st.deepEqual(res, KECCAK256_RLP, 'it has default root')
+      st.end()
+    })
+
+    t.test('should set the state root to empty', async (st) => {
+      const stateManager = new DefaultStateManager({ accountCacheOpts })
+      st.ok(equalsBytes(stateManager._trie.root(), KECCAK256_RLP), 'it has default root')
+
+      // commit some data to the trie
+      const address = new Address(hexStringToBytes('a94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+      const account = createAccount(BigInt(0), BigInt(1000))
+      await stateManager.checkpoint()
+      await stateManager.putAccount(address, account)
+      await stateManager.commit()
+      await stateManager.flush()
+      st.ok(!equalsBytes(stateManager._trie.root(), KECCAK256_RLP), 'it has a new root')
+
+      // set state root to empty trie root
+      await stateManager.setStateRoot(KECCAK256_RLP)
+
+      const res = await stateManager.getStateRoot()
+      st.ok(equalsBytes(res, KECCAK256_RLP), 'it has default root')
+      st.end()
+    })
+
+    t.test('should clear the cache when the state root is set', async (st) => {
+      const stateManager = new DefaultStateManager({ accountCacheOpts })
+      const address = new Address(hexStringToBytes('a94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+      const account = createAccount()
+
+      // test account storage cache
+      const initialStateRoot = await stateManager.getStateRoot()
+      await stateManager.checkpoint()
+      await stateManager.putAccount(address, account)
+
+      const account0 = await stateManager.getAccount(address)
+      st.equal(account0!.balance, account.balance, 'account value is set in the cache')
+
+      await stateManager.commit()
+      const account1 = await stateManager.getAccount(address)
+      st.equal(account1!.balance, account.balance, 'account value is set in the state trie')
+
+      await stateManager.setStateRoot(initialStateRoot)
```

Comment:
I think before this, add a test if the cache is non-empty, and after setting state root then test if the cache is empty.

Reply:
Again, didn't want to mix StateManager API and Cache tests too much.